terminal/command.go

```diff
@@ -199,12 +201,91 @@ func goroutines(client service.Client, args ...string) error {
 	}
 	fmt.Printf("[%d goroutines]\n", len(gs))
 	for _, g := range gs {
-		fmt.Printf("Goroutine %s\n", formatGoroutine(g))
+		prefix := "  "
+		if g.ID == client.EvalScope().GoroutineID {
+			prefix = "* "
+		}
+		fmt.Printf("%sGoroutine %s\n", prefix, formatGoroutine(g))
+	}
+	return nil
+}
+
+func goroutine(client service.Client, args ...string) error {
+	if len(args) == 0 {
+		return printscope(client)
+	}
+	gid, err := strconv.Atoi(args[0])
+	if err != nil {
+		return err
+	}
+
+	oldGid := client.EvalScope().GoroutineID
+
+	newState, err := client.SwitchGoroutine(gid)
+	if err != nil {
+		newState, err = client.GetState()
+		if err != nil {
+			return err
+		}
+	}
+
+	if client.EvalScope().GoroutineID != gid {
+		client.EvalScope().GoroutineID = gid
+		fmt.Printf("Switched from %d to %d (Warning: goroutine is not associated to a thread)\n", oldGid, gid)
+	} else {
+		fmt.Printf("Switched from %d to %d (Thread: %d)\n", oldGid, gid, newState.CurrentThread.ID)
+	}
+
+	return nil
+}
+
+func frame(client service.Client, args ...string) error {
+	if len(args) == 0 {
+		return printscope(client)
 	}
+	var err error
+	client.EvalScope().Frame, err = strconv.Atoi(args[0])
+	return err
+}
+
+func printscope(client service.Client) error {
+	state, err := client.GetState()
+	if err != nil {
+		return err
+	}
+
+	var g *api.Goroutine
+	if state.CurrentGoroutine != nil && state.CurrentGoroutine.ID == client.EvalScope().GoroutineID {
+		g = state.CurrentGoroutine
+	} else {
+		gs, err := client.ListGoroutines()
+		if err != nil {
+			return err
+		}
+
+		for i := range gs {
+			if gs[i].ID == client.EvalScope().GoroutineID {
+				g = gs[i]
+				break
+			}
+		}
+	}
+
+	fmt.Printf("Thread %s\nGoroutine %s Frame %d\n", formatThread(state.CurrentThread), g, client.EvalScope().Frame)
```

Comment:
I think you want `formatGoroutine(g)` here. I'm getting printf formatting errors in the console.